### PR TITLE
Add vscode.d.ts to .eslintignore.

### PR DESCRIPTION
**Pull request by sean-mcmanus**

### CODE REVIEW

File: Extension/.eslintignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 test/**/index.ts
 test/**/runTest.ts
 tools/prepublish.js
+vscode.d.ts
```
Comment (bobbrow) [inline Extension/.eslintignore]: What is vscode.d.ts doing in the root folder?  I thought it should only be in the node_modules folder, or does this pattern match any file with this name throughout the tree?  (and why would eslint be running in the node_modules folder anyway?)

Comment (sean-mcmanus) [reply Extension/.eslintignore]: `"postinstall": "npx vscode-dts dev && npx vscode-dts main"` puts it in the root.

Comment (sean-mcmanus) [reply Extension/.eslintignore]: From https://github.com/microsoft/vscode-dts#readme

Comment (sean-mcmanus) [reply Extension/.eslintignore]: Looks like the node_modules location is the old one, i.e. they have code to delete that.
```typescript
  if (fs.existsSync('node_modules/vscode/vscode.d.ts')) {
    fs.unlinkSync('node_modules/vscode/vscode.d.ts')
```

Comment (bobbrow) [reply Extension/.eslintignore]: Sounds good.  It sounds like it should be in `.vscodeignore` too then, or is it needed for the vsix?

Comment (bobbrow) [reply Extension/.eslintignore]: And use the same pattern as in the `.gitignore` so that both `.d.ts` files are excluded?

Comment (sean-mcmanus) [reply Extension/.eslintignore]: Yeah, good catch, I didn't notice that.

**Review by fearthecowboy — APPROVED** [review]

**Pull request merged**